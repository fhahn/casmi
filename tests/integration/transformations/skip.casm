// transform, cmdline "--fnoskip --fnotrivialblock"
CASM sk

init initR 

// most rules in this file can be reduced to a single skip statement

rule initR=
{
	skip
	skip
	{|
		skip
		{ forall y in [1..3] do skip }
		skip
	|}
	skip
}

function x: -> Int
rule sw =
{
  case x of 
    0: skip
    1: {|
		skip
		skip
		|}
    default: { skip }
  endcase

  case x of 
    0: {skip}
    1: skip
  endcase

  case x of
    default: skip
  endcase
}


rule if_ =
  if x = 3 then {
	skip 
  } else {| skip |}

rule if2_ =
  if x = 3 then
	skip 

rule fa =
	forall y in [1..3] do
		skip

rule already_skip = skip

rule not_possible = {
	skip
	print "Oh noes"
	skip
}
