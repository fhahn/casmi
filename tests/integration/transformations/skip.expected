initR=SKIP
sw=SKIP
if_=SKIP
if2_=SKIP
fa=SKIP
already_skip=SKIP
not_possible=PRINT(LIST(str Oh noes,null),null)
